gematria/datasets/find_accessed_addrs_exegesis.cc

```diff
@@ -134,21 +135,30 @@ Expected<ExecutionAnnotations> ExegesisAnnotator::findAccessedAddrs(
   BenchCode.Key.MemoryValues["memdef1"] = MemVal;
 
   const llvm::MCRegisterInfo &MRI = State.getRegInfo();
+  std::vector<unsigned> UsedRegisters = gematria::getUsedRegisters(
+      *DisInstructions, State.getRegInfo(), State.getInstrInfo());
 
-  for (unsigned i = 0;
-       i < MRI.getRegClass(X86::GR64_NOREX2RegClassID).getNumRegs(); ++i) {
-    RegisterValue RegVal;
-    RegVal.Register =
-        MRI.getRegClass(X86::GR64_NOREX2RegClassID).getRegister(i);
-    RegVal.Value = APInt(64, kInitialRegVal);
-    BenchCode.Key.RegisterInitialValues.push_back(RegVal);
-  }
+  for (unsigned RegisterIndex : UsedRegisters) {
+    RegisterAndValue *NewRegisterValue = MemAnnotations.add_initial_registers();
+    NewRegisterValue->set_register_name(
+        State.getRegInfo().getName(RegisterIndex));
 
-  for (unsigned i = 0; i < MRI.getRegClass(X86::VR128RegClassID).getNumRegs();
-       ++i) {
     RegisterValue RegVal;
-    RegVal.Register = MRI.getRegClass(X86::VR128RegClassID).getRegister(i);
-    RegVal.Value = APInt(128, kInitialRegVal);
+    RegVal.Register = RegisterIndex;
+
+    if (MRI.getRegClass(X86::GR64_NOREX2RegClassID).contains(RegisterIndex)) {
+      RegVal.Value = APInt(64, kInitialRegVal);
+      NewRegisterValue->set_register_value(kInitialRegVal);
+    } else if (MRI.getRegClass(X86::VR128RegClassID).contains(RegisterIndex)) {
+      RegVal.Value = APInt(128, kInitialRegVal);
+      NewRegisterValue->set_register_value(kInitialRegVal);
+    } else if (RegisterIndex == X86::EFLAGS) {
+      RegVal.Value = 0;
+      NewRegisterValue->set_register_value(0);
```

Comment:
Some bits in `EFLAGS` can be set explicitly using separate instructions (e.g. the `DF` flag in question is modified using instructions `STD` and `CLD`). Similar for other flags.

I don't think we need to do anything about it now, but we should keep this in mind, and we might want to set it/allow setting it in the future.

Reply:
Yep. Support would need to be added on the Exegesis side for that. I can look at doing that at some point.